coders/webp.c

```diff
@@ -862,53 +921,107 @@ static MagickBooleanType WriteAnimatedWEBPImage(const ImageInfo *image_info,
   WebPAnimEncoderOptions
     enc_options;
 
-  WebPData
-    webp_data;
-
   WebPPicture
     picture;
 
   WebPAnimEncoderOptionsInit(&enc_options);
   if (image_info->verbose)
     enc_options.verbose = 1;
-
-  image=CoalesceImages(image, exception);
-  first_image=image;
   enc=WebPAnimEncoderNew((int) image->page.width,(int) image->page.height,
     &enc_options);
 
-  head=(PictureMemory *) calloc(sizeof(*head),1);
-  current=head;
+  first_image=CoalesceImages(image,exception);
+  image=first_image;
 
   while (image != NULL)
   {
     if (WebPPictureInit(&picture) == 0)
       ThrowWriterException(ResourceLimitError,"UnableToEncodeImageFile");
 
-    WriteSingleWEBPImage(image_info, image, &picture, current, exception);
+    WriteSingleWEBPPicture(image_info,image,configure,&picture,exception);
     WebPAnimEncoderAdd(enc,&picture,(int) frame_timestamp,configure);
+    WebPPictureFree(&picture);
+
+    if (WebPAnimEncoderGetError(enc) && strlen(WebPAnimEncoderGetError(enc)))
```

Comment:
Do we really need to call `strlen(WebPAnimEncoderGetError(enc)` and I am wondering if we could use the result of the call to `WebPAnimEncoderAdd` to determine if adding the frame failed.